daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -345,14 +337,31 @@ object Node {
     }(recorded, isReplayedBy)
 
   /** Useful in various circumstances -- basically this is what a ledger implementation must use as
-    * a key. The 'hash' should be used when storing the key as value serialization is not guaranteed
-    * to be stable over time (e.g. a new encoding might be introduced).
+    * a key. The 'hash' is guaranteed to be stable over time.
     */
-  sealed abstract case class GlobalKey(templateId: Identifier, key: Value[Nothing], hash: Hash)
+  final class GlobalKey private (
+      val templateId: Identifier,
+      val key: Value[AbsoluteContractId],
```

Comment:
Why the change from `Value[Nothing]` to `Value[AbsoluteContractId]`?

Reply:
the middle plan is to get rid of `RelativeContractId`,  together with the parametrization of  `Value`.  Here the hashing of `key` done at the initialization of the `GlobalKey` object ensure there is no `ContractId` in `key`.